### ISSUE TITLE
flaky test: listening_for_multiple_unspecified_addresses

### ISSUE THREAD

**Issue by koivunej** [rs-ipfs/rust-ipfs#194]
readme-doctest failed on a new flaky test which I had not realized earlier was flaky: `listening_for_multiple_unspecified_addresses` in https://github.com/rs-ipfs/rust-ipfs/pull/192/checks?check_run_id=782193837#step:5:379.

Not sure what do with this, but it's clear the case is similar to `multiple_concurrent_ephemeral_listening_addresses_on_same_ip` on #177.

**Comment by koivunej** [comment]
Fixing this now in the same way as previously. Not closing the issue as the general problem with trying to do it this way still remains.

**Comment by ljedrz** [comment]
I can't reproduce the issue neither with `listening_for_multiple_unspecified_addresses` nor `multiple_concurrent_ephemeral_listening_addresses_on_same_ip` when running the test in a loop.

**Comment by koivunej** [comment]
You have too many cores :) I tried to explain the ordering which is happening in the description. I assume you agree on it's validity?

We could get ton of these errors to pop up if we were able to delay a few operations...

**Comment by ljedrz** [comment]
Yeah, just reporting my experience with these tests; FWIW, I also don't seem to have issues under `#[tokio::test(max_threads = 1)]`.
